### DISCUSSION TITLE
Focus Follow Mouse option for embedded Windows

### DISCUSSION THREAD

**Discussion by Sauermann** [godotengine/godot-proposals#5637] [GUI]
Currently the focus of embedded Windows is changed by clicking on a different embedded window.

Some window managers offer the option, that a window is automatically focused, when the mouse hovers over it.

Is such a feature needed in Godot?

Here is an implementation of the functionality: https://github.com/Sauermann/godot/tree/proposal-focus-follow-mouse

**Comment by theraot** [comment]
For what I can see you made a fork where presumably you implemented the feature you desire. That should a pull request. I believe a proposal where you explain the feature is encourage, yet this discussion should also do. *I believe moderators can move it if necessary.* So, I suggest:

- Open a pull request, mentioning that it implements the feature you mention here (link this discussion). There it can go into the usual review process. *If you believe your implementation is not ready yet, you can make the pull request as a draft.*
- Meanwhile, edit the discussion to explain what the feature is. Consider including a mockup (or - since, I suppose, you have a working implementation - it could be a video) showing the feature.

**Comment by Sauermann** [comment]
I did not find a pull request or proposal fitting, because I do not have an actual problem, that this solves for me.

It was intended as a bug-fix, but unfortunately it ended up not fixing the thing I hoped, it would do. So I put it here for posterity, if the need for this functionality comes up in the future.

**Comment by Calinou** [comment]
> Some window managers offer the option, that a window is automatically focused, when the mouse hovers over it.

Can we query the state of the window manager's focus model somehow, and use it automatically in Godot? Focus-follows-mouse is only really a thing on Linux, and it's not enabled by default in the vast majority of window managers.
